dbt-athena/src/dbt/include/athena/macros/materializations/models/incremental/incremental.sql

```diff
@@ -13,17 +13,14 @@
   {% set temp_schema = config.get('temp_schema') %}
   {% set target_relation = this.incorporate(type='table') %}
   {% set existing_relation = load_relation(this) %}
-  -- If using insert_overwrite on Hive table, allow to set a unique tmp table suffix
-  {% if unique_tmp_table_suffix == True and strategy == 'insert_overwrite' and table_type == 'hive' %}
-    {% set tmp_table_suffix = adapter.generate_unique_temporary_table_suffix() %}
+  -- Generate a unique tmp table suffix if required
+  {% if unique_tmp_table_suffix == True %}
```

Comment:
why did you remove the check on `strategy == 'insert_overwrite' and table_type == 'hive'`?

for sure we can simplify the statement checks and do something like this:
 
```
{% if unique_tmp_table_suffix == True and ((strategy == 'insert_overwrite' and table_type == 'hive') or table_type == 'iceberg') %}
```
But I will avoid to don't check the strategy type.

Reply:
I've some doubts about my comment, your idea might not be bad overall, but to guarantee that works as expected, consider to add some Integrations tests for different scenarios.